code/game/world.dm

```diff
@@ -431,6 +431,24 @@ var/failed_old_db_connections = 0
 		log_world("Feedback database connection established.")
 	return 1
 
+/hook/startup/proc/newRoundNotify()
+	var/people_to_ping = ""
+	var/DBQuery/pull_notify = dbcon.NewQuery("SELECT discord_id FROM [format_table_name("discord")] WHERE notify = 1")
+	if(!pull_notify.Execute())
+		var/err = pull_notify.ErrorMsg()
+		log_game("SQL ERROR while pulling notify people. Error : \[[err]\]\n")
+	else
+		while(pull_notify.NextRow())
+			people_to_ping += "<@" + pull_notify.item[1] + "> "
+	send2mainirc("Server starting up on [station_name()]. Connect to: <byond://[config.server? "[config.server]" : "[world.address]:[world.port]"]> "+people_to_ping, 7355608)
+	send2discord("main", "Server starting up on [station_name()]. Connect to: <byond://[config.server? "[config.server]" : "[world.address]:[world.port]"]> "+people_to_ping, 7355608)
```

Comment:
I think it might be better if, instead of having a 'people to ping' list in-game, you just send2discord a message that can be set, via config, to ping a specific discord usergroup. With that in place, I can set up a role bot in discord that allows people to add/remove this role for themselves. This has several advantages:

- They can opt into / out of notifications without having to log into the game
- Its persistent
- It doesn't require game code to manage
- It enables short new round notifications with at-group-tag rather than at-long-list-of-members

Reply:
The problem with that approach is:
- they can’t notify ingame which was a nice feature
- boxen would need manage roles permissions and need to be recoded to add and remove the role, and automatically remove it on roundstart when the webhook triggers
- if the role is set to mentionable, __anyone__ can mention it, meaning a **LOT** of ping spam can happen

Reply:
1) I guess, I think more people want notifications out of game than in, though.
2) No, it wouldn't. That would be managed by my own, separate bot.
3) Not true. Webhooks can post messages that at-message unmentionable roles, I think.

Reply:
Separate bot is fine, but having the role persist is __not__ user friendly, its a bad idea because its gonna clog recent mentions super fast. As far as I know, there is no way to mention a role that isnt set as mentionable. I have tried force pinging the role ID through a webhook, but while it does show the role mention in chat, it doesnt actually ping the user, so that wouldnt work.

Reply:
Colonial Marines seems to do it in their discord. I'm guessing that's done via webhook. So I assume it must be possible.

Reply:
![image](https://user-images.githubusercontent.com/25063394/58131082-a4a5d680-7c15-11e9-8a68-22970d442cc1.png)
CM has that role pingable by all people. I **REALLY** disagree with this idea as it opens up so many ways to spam people if its a persistent role

Reply:
Hm. When I tried that, it did not come up. So I assumed CM's version was not public pingable.